projects/from-javascript-to-typescript/the-typeinator/02-prototypes-to-classes/index.test.js

```diff
@@ -0,0 +1,162 @@
+				expect(robot.abilities).toEqual(["beep", "navigate"]);
+				expect(robot.power).toBe(100);
+			},
+		);
+
+		(shouldSkip ? test.skip : test)(
+			"announce method logs greetings and abilities",
+			() => {
+				const consoleSpy = jest.spyOn(console, "log").mockImplementation();
+				const robot = new Robot("C-3PO", ["translate", "worry"]);
+
+				robot.announce();
+
+				expect(consoleSpy).toHaveBeenCalledWith("Greetings. I am C-3PO.");
+				expect(consoleSpy).toHaveBeenCalledWith("I am able to translate.");
+				expect(consoleSpy).toHaveBeenCalledWith("I am able to worry.");
+
+				consoleSpy.mockRestore();
+			},
+		);
+
+		(shouldSkip ? test.skip : test)(
+			"charge method increases power and logs status",
+			() => {
+				const consoleSpy = jest.spyOn(console, "log").mockImplementation();
+				const robot = new Robot("Wall-E", ["clean"]);
+				robot.power = 50;
+
+				robot.charge(30);
+
+				expect(robot.power).toBe(80);
+				expect(consoleSpy).toHaveBeenCalledWith(
+					"Recharged power supplies to 80.",
+				);
+
+				consoleSpy.mockRestore();
+			},
+		);
+
+		(shouldSkip ? test.skip : test)(
+			"charge method caps power at 100 and logs optimal capacity",
+			() => {
+				const consoleSpy = jest.spyOn(console, "log").mockImplementation();
+				const robot = new Robot("BB-8", ["roll"]);
+				robot.power = 90;
+
+				robot.charge(20);
+
+				expect(robot.power).toBe(100);
+				expect(consoleSpy).toHaveBeenCalledWith(
+					"Recharged power supplies to 100.",
+				);
+				expect(consoleSpy).toHaveBeenCalledWith(
+					"I am at optimal operational capacity.",
+				);
+
+				consoleSpy.mockRestore();
+			},
+		);
+
+		(shouldSkip ? test.skip : test)(
+			"move method reduces power and logs movement",
+			() => {
+				const consoleSpy = jest.spyOn(console, "log").mockImplementation();
+				const robot = new Robot("EVE", ["scan"]);
+
+				robot.move(25);
+
+				expect(robot.power).toBe(75);
+				expect(consoleSpy).toHaveBeenCalledWith("Moving 25 units.");
+
+				consoleSpy.mockRestore();
+			},
+		);
+
+		(shouldSkip ? test.skip : test)(
+			"move method prevents movement when insufficient power",
+			() => {
+				const consoleSpy = jest.spyOn(console, "log").mockImplementation();
+				const robot = new Robot("WALL-E", ["compact"]);
+				robot.power = 10;
+
+				robot.move(25);
+
+				expect(robot.power).toBe(10);
+				expect(consoleSpy).toHaveBeenCalledWith(
+					"I do not have enough power to move 25 units.",
+				);
+
+				consoleSpy.mockRestore();
+			},
+		);
+	});
+
+	describe("Humanoid", () => {
+		(shouldSkip ? test.skip : test)(
+			"creates a humanoid with name, abilities, power, and catchphrase",
+			() => {
+				const humanoid = new Humanoid(
+					"Data",
+					["compute", "analyze"],
+					"Fascinating",
+				);
+
+				expect(humanoid.name).toBe("Data");
+				expect(humanoid.abilities).toEqual(["compute", "analyze"]);
+				expect(humanoid.power).toBe(100);
+				expect(humanoid.catchphrase).toBe("Fascinating");
```

Comment:
[Testing] The four individual assertions work, but they only test one thing at a time. Only one failure will be shown at a time. Using a full `.toEqual()` will get all failures at once, and is less code:

```suggestion
		expect(humanoid).toEqual({
			abilities: ["compute", "analyze"],
			catchphrase: "Fascinating",
			name: "Data",
			power: 100,
		});
```